### DISCUSSION TITLE
Historical data not showing up in InfluxDB dashboard

### DISCUSSION THREAD

**Discussion by HyruleWarrior** [jertel/vuegraf#134] [Q&A] [answered]
I applied the `influx_dashboard.json` dashboard to my new `influxdb2` and pulled in historical data adding these commands to entry point:
```
-- resetdatabase --historydays=720 /opt/vuegraf/conf/vuegraf.json
```

It appears the historical data import succeeds from logs (although I think when I first did this earlier today before starting fresh it said there were 14k data points, not sure what changed there):
```
2023-09-07 19:51:44.497478 | INFO  | Loading historical data: 720 day(s) ago
2023-09-07 19:55:03.465759 | INFO  | Submitting datapoints to database; account="Main Panel"; points=9743
2023-09-07 19:56:03.945817 | INFO  | Submitting datapoints to database; account="Main Panel"; points=18
```

But when looking at the dashboard I am only seeing new data:
<img width="2453" alt="image" src="https://github.com/jertel/vuegraf/assets/5056766/67a22056-2d69-48c5-8c21-5007faea72c6">

Any ideas why historical data isn't showing? I would expect ~1 year of data as that is when I set up my Emporia and that is what I can see via the app.

**Comment by jertel** [comment]
My guess is because the the history data is labeled as detailed=Day or detailed=Hour, but the provided dashboard is more intended for future data visualization. Its "DetailedDataEnabled" dropdown only contains True or False, and lacks the Day and Hour options.

You could try adding those two options via the Settings screen.
![image](https://github.com/jertel/vuegraf/assets/16639611/f7a5e5bd-8e95-4416-9945-5ff1c50c2b43)

 @gauthig authored a large portion of the historic data fetch and might have some additional insights. See #119 for more background.

**Reply by gauthig** [reply, marked as answer]
Yes, it seems to do with the selection of Detailed.  What I did was clone the graph twice, then in one I selected Detailed = Day and setup for 365 days as a static range.  Then on the other one I set Detailed = Hour and set the static range for Days - 7.  
This way I have three graphs:
Top - Showing whatever range I selected (usually 2 days) which has minutes (I do not see much in seconds as energy use = cost over time)
Middle - Showing the current week in hourly values
Bottom - Quick easy to read annual graph by days.
